cli/bpmetadata/tfconfig_test.go

```diff
@@ -266,3 +266,36 @@ func TestTFRoles(t *testing.T) {
 		})
 	}
 }
+
+func TestTFProviderVersions(t *testing.T) {
+	tests := []struct {
+		name                 string
+		configName           string
+		wantProviderVersions []*ProviderVersion
+	}{
+		{
```

Comment:
Could we add a few more cases
- no provider versions at all
- only source and no versions

Reply:
Yeah TF would complain and our linter would fail for incomplete/invalid. The valid cases to test here are
- no provider info (TF downloads implied latest according to constraints in child modules if any) - example https://github.com/terraform-google-modules/terraform-google-network/blob/master/codelabs/simple/main.tf
- source is mentioned but no version - example https://github.com/terraform-google-modules/terraform-google-container-vm/blob/b9c7fdd2cd0bd09942440ac372fdee47bde57db9/examples/instance_with_config_file/versions.tf#L19-L21

Reply:
Tests added and all passing on local runs.